src/Nethermind/Nethermind.Synchronization.Test/Mocks/FirstFree.cs

```diff
@@ -31,9 +31,9 @@ private FirstFree()
 
         public bool CanBeReplaced => false;
 
-        public PeerInfo Allocate(PeerInfo currentPeer, IEnumerable<PeerInfo> peers, INodeStatsManager nodeStatsManager, IBlockTree blockTree)
+        public PeerInfo Allocate(PeerInfo? currentPeer, IEnumerable<PeerInfo> peers, INodeStatsManager nodeStatsManager, IBlockTree blockTree)
         {
-            return peers.FirstOrDefault() ?? currentPeer;
+            return peers.FirstOrDefault() ?? currentPeer!;
```

Comment:
Doesn't make much sense to me

Reply:
Either peers has an item or has a currentPeer?